core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -1333,6 +1333,34 @@ object GroupCoordinator {
     apply(config, zkClient, replicaManager, heartbeatPurgatory, joinPurgatory, time, metrics)
   }
 
+  def apply(config: KafkaConfig,
```

Comment:
This looks very similar to the previous `apply` function.  Can we express the one in terms of the other to avoid code duplication?